File: app/src/main/java/me/gabeg/sicksends/SsTopRopeRepository.java
package me.gabeg.sicksends;

import android.app.Application;

/**
 * Repository for top rope problems.
 */
public class SsTopRopeRepository
	extends SsProblemRepository<SsTopRopeDao, SsTopRope>
{

	/**
	 */
	public SsTopRopeRepository(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemRepository#buildDao(SsProblemDatabase)
	 */
	@Override
	protected void buildDao(SsProblemDatabase db)
	{
		this.mProblemDao = db.topRopeDao();
	}

}

File: app/src/main/java/me/gabeg/sicksends/db/toprope/SsTopRopeRepository.kt
package me.gabeg.sicksends.db.toprope

import me.gabeg.sicksends.db.generic.SsGenericProblemRepository
import javax.inject.Inject

/**
 * Repository for top rope problems.
 */
class SsTopRopeRepository @Inject constructor(dao : SsTopRopeDao)
	: SsGenericProblemRepository<SsTopRopeProblem>(dao)
